src/vsphere_cpi/lib/cloud/vsphere/resources/network.rb

```diff
@@ -0,0 +1,98 @@
+require 'cloud/vsphere/resources'
+
+module VSphereCloud
+  module Resources
+    class Network
+      include VimSdk
+      include ObjectStringifier
+      stringify_with :name, :mob, :bosh_name
+
+      attr_reader :mob, :client, :bosh_name
+      def self.make_network_resource(bosh_name, mob, client)
+        case mob
```

Comment:
I think the `case` statement should return the type. That way we don't repeat the `.new(bosh_name, mob, client)` clause.